.github/workflows/run-tests.yml

```diff
@@ -20,7 +15,40 @@ on:
         default: ${{ github.sha }}
 
 jobs:
-  do-nothing:
-    runs-on: ubuntu-latest
+  unit-tests:
+    strategy:
+      # By default, if any job in a matrix fails, all other jobs are immediately canceled. This makes the jobs run
+      # to completion instead.
+      fail-fast: false
+      matrix:
+        node: ${{ fromJson(inputs.node-matrix) }}
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
     steps:
-      - run: echo DOING NOTHING FOR NOW
+      # Check out the code
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.target-branch }}
+      # Make sure we're on the right version on Node
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node.version }}
+      - run: yarn
+      - run: yarn build
+      - run: yarn test
+        id: jest-tests
+      - run: yarn lint
+      - if: ${{ failure() && steps.jest-tests.outcome != 'success' }}
+        # Use bash for this to avoid any syntax nonsense
+        shell: bash
+        run: |
+          MSG="Jest tests failed. If no annotations were added, the most likely problem is insufficient code coverage."
+          echo "::error ::$MSG"
```

Comment:
Can you help me understand why we need this? Is this just to get a better error message when code coverage is insufficient? Is it really needed? Doesn't yarn test (which runs jest --coverage) already take of all this with a sufficient error message?

Reply:
Looking at this PR job... it actually failed the test because of zero coverage and the message there is pretty clean. I do don't think we need this additional step.

You may want to add in a dummy test for now to satisfy coverage (which is what I did when I started from an empty repo).

Lastly, you might need/want to put in "esModuleInterp": true in your base tsconfig to satisfy jest in some cases.

Reply:
Under the old pattern, I experienced people being confused when coverage was the problem, because the  tests failed but the lack of an annotation made it not immediately clear why.
For now I feel like it's fine to leave the coverage where it is, and part of the next PR will be enforcing it as a requirement for merging.
Also, thanks for the heads up; I'll look into `esModuleInterop`.